test/cpp/jit/test_lite_interpreter.cpp

```diff
@@ -454,21 +488,125 @@ void testLiteInterpreterSequentialModuleInfo() {
   std::unordered_set<std::string> module_debug_info_set;
   size_t pc = 0;
   while (true) {
-    std::string module_info = bc.get_forward_method_debug_info(pc);
-    if (module_info == "<no module info for pc>") {
+    // std::string module_info = bc.get_forward_method_debug_info(pc);
+    // if (module_info == "<no module info for pc>") {
+    //   break;
+    // }
+    // if (!module_info.empty()) {
+    //   module_debug_info_set.insert(module_info);
+    // }
+    // ++pc;
+    try {
+      std::string module_info = bc.get_forward_method_debug_info(pc);
+      if (!module_info.empty() && module_info != "<no module info>") {
+        module_debug_info_set.insert(module_info);
+      }
+      ++pc;
+    } catch (const std::exception& e) {
       break;
     }
-    if (!module_info.empty()) {
-      module_debug_info_set.insert(module_info);
-    }
-    ++pc;
   }
 
   std::unordered_set<std::string> expected_result(
       {"top(C).A0(A).forward", "top(C).B0(B).forward"});
   AT_ASSERT(module_debug_info_set == expected_result);
 }
 
+void testLiteInterpreterHierarchyModuleInfo() {
+  Module a("A");
+  a.define(R"JIT(
+    def forward(self, x):
+      return x + 1
+  )JIT");
+  Module b("B");
+  b.register_module("A0", a);
+  b.define(R"JIT(
+    def forward(self, x):
+      return self.A0.forward(x) + 1
+  )JIT");
+  Module c("C");
+  c.register_module("B0", b);
+  c.define(R"JIT(
+    def forward(self, x):
+      return self.B0.forward(x) + 1
+  )JIT");
+
+  std::stringstream ss;
+  c._save_for_mobile(ss, {}, true);
+  mobile::Module bc = _load_for_mobile(ss);
+
+  std::unordered_set<std::string> module_debug_info_set;
+  size_t pc = 0;
+  while (true) {
+    // std::string module_info = bc.get_forward_method_debug_info(pc);
+    // if (module_info == "<no module info for pc>") {
+    //   break;
+    // }
+    // if (!module_info.empty()) {
+    //   module_debug_info_set.insert(module_info);
+    // }
+    // ++pc;
+    try {
+      std::string module_info = bc.get_forward_method_debug_info(pc);
+      if (!module_info.empty() && module_info != "<no module info>") {
+        module_debug_info_set.insert(module_info);
+      }
+      ++pc;
+    } catch (const std::exception& e) {
+      break;
+    }
+  }
+
+  std::unordered_set<std::string> expected_result(
```

Comment:
Thanks for adding this test. I think it might help to just add a comment explaining why 3 nodes. Each for each of the add. And each add occurs inside a different module.